crates/api/src/fetch.rs

```diff
@@ -0,0 +1,34 @@
+//! Kitsune2 fetch types.
+
+use std::sync::Arc;
+
+use crate::{builder, config, AgentId, BoxFut, K2Result, OpId};
+
+/// Trait for implementing a fetch module to fetch ops from other agents.
+pub trait Fetch: 'static + Send + Sync {
```

Comment:
Please add the debug requirement. When working with trait objects it's important to be able to get *some* information out of it when testing / tracing.